src/workerd/api/actor.c++

```diff
@@ -23,18 +23,18 @@ public:
   kj::Own<WorkerInterface> newSingleUseClient(kj::Maybe<kj::String> cfStr) override {
     auto& context = IoContext::current();
 
-    // Lazily initialize actorChannel
-    if (actorChannel == nullptr) {
-      auto& context = IoContext::current();
-      actorChannel = context.getColoLocalActorChannel(channelId, actorId);
-    }
-
     return context.getMetrics().wrapActorSubrequestClient(context.getSubrequest(
         [&](SpanBuilder& span, IoChannelFactory& ioChannelFactory) {
       if (span.isObserved()) {
         span.setTag("actor_id"_kjc, kj::str(actorId));
       }
 
+      // Lazily initialize actorChannel
+      if (actorChannel == nullptr) {
+        auto& context = IoContext::current();
```

Comment:
Is there any chance this will be a different `IoContext` than the one associated with the `context.getMetrics()` call above? If so, does that matter? If it does, perhaps it would be better to use the captured `context` rather than calling `IoContext::current()` here?

Reply:
Oh, yeah, that's a good point.  Going through the implementation of `IoContext::getSubrequest()`, it looks like if the function gets called at all, it is called synchronously, so the context should be identical.

Removed the redundant `IoContext::current()` call, and updated the `IoContext::getSubrequest()` doc comment to clarify its contract.